examples/acoustic/acoustic_example.py

```diff
@@ -42,7 +42,7 @@ def run(dimensions=(150, 150, 50), spacing=(20.0, 20.0, 20.0), tn=250.0,
     # Smooth velocity
     initial_vp = smooth10(true_vp, dimensions)
 
-    dm = true_vp**-2 - initial_vp**-2
```

Comment:
Ok two points here:
a) This is a general design question that warrants a separate github issue. I can see points for and against making Devito-internal allocation optional.
b) For the benchmarking examples this should be fixed regardless of that discussion, since the startup time can limit our benchmarking with _really big_ grids. The fix would effectively involve deferring the initialisation of `m` by separating the code that generates the `vp` and passing it into the `Forward` operator initialisation routines.